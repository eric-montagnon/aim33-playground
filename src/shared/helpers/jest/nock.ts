import { env } from "@src/env";
import nock from "nock";

export const nockWithEndpoint = (options?: nock.Options) =>
  nock(env.API_ENDPOINT, options);
