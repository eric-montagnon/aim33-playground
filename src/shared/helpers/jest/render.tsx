import { ThemeProvider } from "@emotion/react";
import { queryClient } from "@src/shared/infra/clients/queryClient";

import { QueryBoundaries } from "@src/shared/view/components/QueryBoundaries/QueryBoundaries.component";
import { ToastService } from "@src/shared/view/services/ToastService/Toast.service";
import { theme } from "@src/shared/view/theme/theme";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  RenderOptions,
  // eslint-disable-next-line testing-library/no-manual-cleanup
  cleanup,
  render,
  screen,
  waitFor,
} from "@testing-library/react-native";
import React from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

// This may seem like a hack but it's reliable. afterEach hooks always run when we expect them too
let afterEachHook: (() => void) | undefined = undefined;
afterEach(() => {
  afterEachHook?.();
  afterEachHook = undefined;
});

export const wrapAndRender = (
  component: React.ReactElement,
  options?: RenderOptions,
) => {
  const frame = { x: 0, y: 0, width: 0, height: 0 };
  const insets = { top: 0, left: 0, right: 0, bottom: 0 };

  // In case there are multiple `renderWithQueries` calls in the same test, cleanup
  afterEachHook?.();

  // TODO: clean reusable solution logging
  // The idea is that in tests, you don't want to log known errors (you often write tests for these so it would be annoying)
  // But you want to log unknown errors, otherwise debugging tests is hell
  // setLogger(/* */);

  const waitForQueries = async () => {
    await waitFor(() => {
      assertNoPendingQueries(queryClient);
    });
  };

  afterEachHook = () => {
    // React cleanup should happen before waiting for queries, otherwise strange things happen
    cleanup();

    // Pending queries when a test end can cause multiple problems
    // - Not testing what you think you do (loaders in snapshots...)
    // - Test independence problems
    // - "Jest did not exit..." errors
    // -> This is important
    assertNoPendingQueries(queryClient);

    // This is a workaround for https://github.com/tannerlinsley/react-query/issues/1847
    queryClient.clear();
  };

  const renderResult = render(component, {
    wrapper: ({ children }) => (
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider
          initialMetrics={{
            frame,
            insets,
          }}
        >
          <ThemeProvider theme={theme}>
            <QueryBoundaries>{children}</QueryBoundaries>
            <ToastService.Toast />
          </ThemeProvider>
        </SafeAreaProvider>
      </QueryClientProvider>
    ),
    ...options,
  });

  return {
    ...renderResult,
    /**
     * When it's too complicated / not important for a test to wait for pending queries by checking UI elements
     * This function can be used as a fallback
     */
    waitForQueries,
  };
};

afterEach(() => {
  queryClient.clear();
});

export const waitForLoaders = () => {
  return waitFor(() => {
    const loader = screen.queryByTestId("activity-indicator");
    if (loader) throw new Error("Loader still visible");
  });
};

const assertNoPendingQueries = (queryClient: QueryClient) => {
  // Not using queryClient.isFetching() / isMutating() because we want to log details on the queries
  // const pendingQueries = queryClient
  //   .getQueryCache()
  //   .findAll(undefined)
  //   // The `filter` param of `.findAll` seems to be unreliable, so we filter manually (v3.16.0 de react-query, might have changed that)
  //   .filter((query) => query.isFetching())
  //   .map((query) => JSON.stringify(query.queryKey));
  const numberOfPendingQueries = queryClient.isFetching();

  if (numberOfPendingQueries > 0) {
    throw new Error(`Some queries still pending:\n${numberOfPendingQueries}`);
  }

  const pendingMutations = queryClient
    .getMutationCache()
    .findAll({})
    // The `filter` param of `.findAll` seems to be unreliable, so we filter manually (v3.16.0 de react-query, might have changed that)
    .filter((mutation) => mutation.state.status === "loading");

  if (pendingMutations.length > 0) {
    throw new Error(`${pendingMutations.length} mutations still pending`);
  }
};
