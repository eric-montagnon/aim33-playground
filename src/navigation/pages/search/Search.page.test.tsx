import { screen, waitFor } from "@testing-library/react-native";
import * as React from "react";

import { drinksListMock } from "@src/modules/drinks/drink.mock";
import { SearchScreenProps } from "@src/navigation/Navigator.interface";
import { SearchPage } from "@src/navigation/pages/search/Search.page";
import { wrapAndRender } from "@src/shared/helpers/jest/render";
import nock from "nock";

const navigateMock = jest.fn();
const searchTestProps = {
  navigation: {
    navigate: navigateMock,
  },
} as unknown as SearchScreenProps;

describe("Sign up Page", () => {
  it("displays properly", async () => {
    const scope = nock("https://www.thecocktaildb.com/api/json/v1/1/search.php")
      .get("?s=punch")
      .reply(200, { drinks: drinksListMock });

    wrapAndRender(<SearchPage {...searchTestProps} />);

    await waitFor(() => expect(screen.getByText("GG")).toBeTruthy());

    expect(scope.isDone()).toBeTruthy();
    expect(screen).toMatchComponentSnapshot();
  });
});
