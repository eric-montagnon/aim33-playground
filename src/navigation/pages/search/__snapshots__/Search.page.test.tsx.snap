// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Page displays properly 1`] = `
"https://cdn-icons-png.flaticon.com/512/3076/3076134.png
ó°…–

https://www.thecocktaildb.com/images/media/drink/vyxwut1468875960.jpg
GG

https://www.thecocktaildb.com/images/media/drink/2x8thr1504816928.jpg
A1

https://www.thecocktaildb.com/images/media/drink/l3cd7f1504818306.jpg
Ace


"
`;

exports[`Sign up Page displays properly 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "height": 0,
        }
      }
    />
    <RCTScrollView
      alwaysBounceHorizontal={false}
      alwaysBounceVertical={false}
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
      style={{}}
    >
      <View>
        <View
          style={
            {
              "padding": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "lightgrey",
                "borderRadius": 28,
                "margin": 24,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="search-bar-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "lightgrey",
                  "borderRadius": 28,
                  "flex": undefined,
                  "flexDirection": "row",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="search-bar-container"
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 6,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="search-bar-icon-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="search-bar-icon-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityLabel="search"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": true,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityTraits="button"
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    style={
                      {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "overflow": "hidden",
                      }
                    }
                    testID="search-bar-icon"
                  >
                    <Image
                      accessibilityElementsHidden={true}
                      accessibilityIgnoresInvertColors={true}
                      importantForAccessibility="no-hide-descendants"
                      source={
                        {
                          "uri": "https://cdn-icons-png.flaticon.com/512/3076/3076134.png",
                        }
                      }
                      style={
                        {
                          "height": 24,
                          "resizeMode": "contain",
                          "tintColor": "rgba(73, 69, 79, 1)",
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                          "width": 24,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <TextInput
                accessibilityRole="search"
                keyboardAppearance="light"
                placeholder="Search"
                placeholderTextColor="rgba(28, 27, 31, 1)"
                returnKeyType="search"
                selectionColor="rgba(103, 80, 164, 1)"
                style={
                  {
                    "alignSelf": "stretch",
                    "color": "rgba(73, 69, 79, 1)",
                    "flex": 1,
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 0,
                    "minHeight": 56,
                    "minWidth": 0,
                    "paddingLeft": 0,
                    "textAlign": "left",
                  }
                }
                testID="search-bar"
                underlineColorAndroid="transparent"
                value="punch"
              />
              <View
                pointerEvents="auto"
                style={{}}
                testID="search-bar-icon-wrapper"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "borderRadius": 20,
                      "height": 40,
                      "margin": 6,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                      "width": 40,
                    }
                  }
                  testID="search-bar-clear-icon-container-outer-layer"
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "transparent",
                        "borderColor": "rgba(121, 116, 126, 1)",
                        "borderRadius": 20,
                        "borderWidth": 0,
                        "elevation": 0,
                        "flex": 1,
                        "overflow": "hidden",
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                      }
                    }
                    testID="search-bar-clear-icon-container"
                  >
                    <View
                      accessibilityComponentType="button"
                      accessibilityLabel="clear"
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityTraits="button"
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      centered={true}
                      collapsable={false}
                      focusable={true}
                      hitSlop={
                        {
                          "bottom": 6,
                          "left": 6,
                          "right": 6,
                          "top": 6,
                        }
                      }
                      style={
                        {
                          "alignItems": "center",
                          "borderRadius": 20,
                          "flexGrow": 1,
                          "justifyContent": "center",
                          "overflow": "hidden",
                        }
                      }
                      testID="search-bar-clear-icon"
                    >
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        selectable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "color": "rgba(73, 69, 79, 1)",
                            "fontFamily": "material-community",
                            "fontSize": 24,
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          }
                        }
                      >
                        ó°…–
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "backgroundColor": "#574BAA",
                "borderRadius": 10,
                "paddingHorizontal": 10,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
          >
            <View
              style={
                {
                  "height": 10,
                }
              }
            />
            <View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                style={
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "padding": 10,
                    "shadowColor": "#000000",
                    "shadowOffset": {
                      "height": 3,
                      "width": 0,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 2,
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "https://www.thecocktaildb.com/images/media/drink/vyxwut1468875960.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": 70,
                      "width": 70,
                    }
                  }
                />
                <View
                  style={
                    {
                      "width": 10,
                    }
                  }
                />
                <Text
                  numberOfLines={0}
                  style={
                    {
                      "color": "#000000",
                      "fontFamily": "OpenSans-Bold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textDecorationLine": "none",
                    }
                  }
                >
                  GG
                </Text>
              </View>
              <View
                style={
                  {
                    "height": 10,
                  }
                }
              />
            </View>
            <View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                style={
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "padding": 10,
                    "shadowColor": "#000000",
                    "shadowOffset": {
                      "height": 3,
                      "width": 0,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 2,
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "https://www.thecocktaildb.com/images/media/drink/2x8thr1504816928.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": 70,
                      "width": 70,
                    }
                  }
                />
                <View
                  style={
                    {
                      "width": 10,
                    }
                  }
                />
                <Text
                  numberOfLines={0}
                  style={
                    {
                      "color": "#000000",
                      "fontFamily": "OpenSans-Bold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textDecorationLine": "none",
                    }
                  }
                >
                  A1
                </Text>
              </View>
              <View
                style={
                  {
                    "height": 10,
                  }
                }
              />
            </View>
            <View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                style={
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "padding": 10,
                    "shadowColor": "#000000",
                    "shadowOffset": {
                      "height": 3,
                      "width": 0,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 2,
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "https://www.thecocktaildb.com/images/media/drink/l3cd7f1504818306.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": 70,
                      "width": 70,
                    }
                  }
                />
                <View
                  style={
                    {
                      "width": 10,
                    }
                  }
                />
                <Text
                  numberOfLines={0}
                  style={
                    {
                      "color": "#000000",
                      "fontFamily": "OpenSans-Bold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textDecorationLine": "none",
                    }
                  }
                >
                  Ace
                </Text>
              </View>
              <View
                style={
                  {
                    "height": 10,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            accessibilityLabel="Go Back"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            style={
              {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="24"
              bbWidth="24"
              fill="none"
              focusable={false}
              height={24}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                }
              }
              testID="left-arrow-icon"
              vbHeight={24}
              vbWidth={24}
              width={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M17 11H9.41l3.3-3.29a1.004 1.004 0 1 0-1.42-1.42l-5 5a1 1 0 0 0-.21.33 1 1 0 0 0 0 .76 1 1 0 0 0 .21.33l5 5a1.002 1.002 0 0 0 1.639-.325 1 1 0 0 0-.22-1.095L9.41 13H17a1 1 0 0 0 0-2Z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          style={
            {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateY": -0,
          },
        ],
      }
    }
    testID="toastAnimatedContainer"
  >
    <View
      style={
        {
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EBFBF1",
            "flex": 1,
            "flexDirection": "row",
            "padding": 16,
            "paddingTop": 16,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "flex": 0,
              "height": 24,
              "width": 24,
            }
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M18.7104 7.20998C18.6175 7.11625 18.5069 7.04186 18.385 6.99109C18.2632 6.94032 18.1324 6.91418 18.0004 6.91418C17.8684 6.91418 17.7377 6.94032 17.6159 6.99109C17.494 7.04186 17.3834 7.11625 17.2904 7.20998L9.84044 14.67L6.71044 11.53C6.61392 11.4367 6.49998 11.3634 6.37512 11.3142C6.25026 11.265 6.11694 11.2409 5.98276 11.2432C5.84858 11.2455 5.71617 11.2743 5.59309 11.3278C5.47001 11.3812 5.35868 11.4585 5.26544 11.555C5.1722 11.6515 5.09889 11.7654 5.04968 11.8903C5.00048 12.0152 4.97635 12.1485 4.97867 12.2827C4.98099 12.4168 5.00972 12.5493 5.06321 12.6723C5.1167 12.7954 5.19392 12.9067 5.29044 13L9.13044 16.84C9.2234 16.9337 9.334 17.0081 9.45586 17.0589C9.57772 17.1096 9.70843 17.1358 9.84044 17.1358C9.97245 17.1358 10.1032 17.1096 10.225 17.0589C10.3469 17.0081 10.4575 16.9337 10.5504 16.84L18.7104 8.67998C18.8119 8.58634 18.893 8.47269 18.9484 8.34619C19.0038 8.21969 19.0324 8.08308 19.0324 7.94498C19.0324 7.80688 19.0038 7.67028 18.9484 7.54378C18.893 7.41728 18.8119 7.30363 18.7104 7.20998V7.20998Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 8,
            }
          }
        >
          <Text
            accessibilityRole="alert"
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 20,
                "lineHeight": 30,
                "textDecorationLine": "none",
              }
            }
          />
          <Text
            accessibilityRole="alert"
            numberOfLines={0}
            style={
              {
                "color": "#000000",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textDecorationLine": "none",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
